packages/chaire-lib-common/src/tasks/dataImport/data/dataGeojson.ts

```diff
@@ -67,3 +70,64 @@ export class DataFileGeojson extends DataGeojson {
         return this._fileData || [];
     }
 }
+
+// Instead of reading the entire file at once, this class streams it asynchronously. This allows for large files to be read without crashing the application.
+export class DataStreamGeojson extends DataGeojson {
+    private _fileData: GeoJSON.Feature[] | undefined = undefined;
+    private _filename: string;
+    private _dataInitialized: boolean;
+
+    constructor(filename: string) {
+        super({ type: 'FeatureCollection', features: [] });
+        this._filename = filename;
+        this._dataInitialized = false;
+    }
+
+    // Factory method so that we can create the class while calling an async function.
```

Comment:
Can we elaborate a bit more in the comment here to explain that this is a async/not async work around to avoid having to rewrite everything. 
Maybe in a TODO: form